### PR TITLE
Add SIGUSR2 pprof to agent and proxy

### PR DESCRIPTION
**Pull request by jasonodonnell**
This adds the ability to get pprof data from Agent and Proxy using SIGUSR2, similar to Vault server.

### CODE REVIEW

File: command/agent.go [hunk 1]
```diff
@@ -758,6 +761,47 @@ func (c *AgentCommand) Run(args []string) int {
 				case c.reloadedCh <- struct{}{}:
 				default:
 				}
+			case <-c.SighUSR2Ch:
+				// We can only get pprof outputs via the API but sometimes Vault can get
+				// into a state where it cannot process requests so we can get pprof outputs
+				// via SIGUSR2.
```
Comment (VioletHynes) [inline command/agent.go:767]:
```suggestion
```

Comment (VioletHynes) [reply command/agent.go:767]: I don't believe this is accurate for Agent/Proxy:

> We can only get pprof outputs via the API

**Review by VioletHynes — APPROVED** [review]
This looks good! We should add this to Proxy too -- if it's not too much to ask :)

File: command/proxy.go [hunk 2]
```diff
@@ -715,6 +718,44 @@ func (c *ProxyCommand) Run(args []string) int {
 				case c.reloadedCh <- struct{}{}:
 				default:
 				}
+			case <-c.SigUSR2Ch:
+				pprofPath := filepath.Join(os.TempDir(), "vault-agent-pprof")
```
Comment (VioletHynes) [inline command/proxy.go]:
```suggestion
				pprofPath := filepath.Join(os.TempDir(), "vault-proxy-pprof")
```

**Comment by jasonodonnell** [comment]
> This looks good! We should add this to Proxy too -- if it's not too much to ask :)

Added!

File: changelog/27510.txt [hunk 3]
```diff
@@ -0,0 +1,3 @@
+```release-note:improvement
+agent: Add the ability to dump pprof to the filesystem using SIGUSR2
+```
```
Comment (VioletHynes) [inline changelog/27510.txt:3]:
```suggestion
```release-note:improvement
agent: Add the ability to dump pprof to the filesystem using SIGUSR2
```

```release-note:improvement
proxy: Add the ability to dump pprof to the filesystem using SIGUSR2
```
```

Comment (VioletHynes) [reply changelog/27510.txt:3]: okay the suggestion got a little messed up with the triple backticks but we should essentially have two of these, one for Agent, one for Proxy

They can be identical other than the component

**Review by VioletHynes — COMMENTED** [review]
Looks great! Two small suggestions (changelog for proxy, and update to `vault-proxy-pprof`) but this looks great. Thanks for adding this!

**Pull request merged**